front/pages/api/v1/w/[wId]/apps/[aId]/runs/index.ts

```diff
@@ -264,92 +186,194 @@ async function handler(
         });
       }
 
-      await Run.create({
-        dustRunId: runRes.value.run.run_id,
-        appId: app.id,
-        runType: "deploy",
-        workspaceId: keyRes.value.workspaceId,
-      });
-
-      let run: RunType = runRes.value.run;
-      run.specification_hash = run.app_hash;
-      delete run.app_hash;
-
-      // If `blocking` is set, poll for run completion.
-      if (req.body.blocking) {
-        const runId = run.run_id;
-        try {
-          await poll({
-            fn: async () => {
-              const run = await coreAPI.getRunStatus({
-                projectId: app.dustAPIProjectId,
-                runId,
+      if (req.body.stream) {
+        // Start SSE stream.
+        res.writeHead(200, {
+          "Content-Type": "text/event-stream",
+          "Cache-Control": "no-cache",
+          Connection: "keep-alive",
+        });
+      } else if (!req.body.blocking) {
+        // Non blocking, return a run object as soon as we get the runId.
+        void (async () => {
+          try {
+            const dustRunId = await runRes.value.dustRunId;
+
+            const statusRunRes = await coreAPI.getRunStatus({
+              projectId: app.dustAPIProjectId,
+              runId: dustRunId,
+            });
+
+            if (statusRunRes.isErr()) {
+              return apiError(req, res, {
+                status_code: 400,
+                api_error: {
+                  type: "run_error",
+                  message: "There was an error getting the app run status.",
+                  run_error: statusRunRes.error,
+                },
               });
-              if (run.isErr()) {
-                return { status: "error" };
-              }
-              const r = run.value.run;
-              return { status: r.status.run };
-            },
-            validate: (r) => {
-              if (r && r.status == "running") {
-                return false;
+            }
+
+            const run: RunType = statusRunRes.value.run;
+            run.specification_hash = run.app_hash;
+            delete run.app_hash;
+
+            run.status.blocks = [];
+            run.results = null;
+
+            res.status(200).json({ run: run as RunType });
+          } catch (err) {
```

Comment:
This catch is not needed. rgetRunStatus returns a Res and any throw in there should be a 500 not a 400